src/libclipper/include/clipper/selection_policies.hpp

```diff
@@ -8,324 +8,123 @@
 #include "datatypes.hpp"
 #include "task_executor.hpp"
 
-/**
- * IMPORTANT NOTES FOR USING SELECTION POLICIES
- * The selection policy only supports binary classfication models
- * The binary classes must use 0 as negative class indicator and 1 as positive
- * class indicator
- */
-
 namespace clipper {
 
 const std::string LOGGING_TAG_SELECTION_POLICY = "SELECTIONPOLICY";
 
-// *********
-// * State *
-// *********
-
-/**
- * Model Information
- * Each model has properties and we use an unordered_map to contain these
- * properties
- *  - EXP3/EXP4 Model Properties:
- *    "weight": weight of this model
- *  - EpsilonGreedy/UCB Properties:
- *    "expected_loss": the mean of the loss distribution of this model
- *    "times_selected": how many times we have used this model
- */
-using ModelInfo = std::unordered_map<std::string, double>;
-
-/** Model Map
- * A map of the models to their corresponding model information
- */
-using Map = std::unordered_map<VersionedModelId, ModelInfo,
-                               std::function<size_t(const VersionedModelId&)>>;
-
-class BanditPolicyState {
+class SelectionState {
  public:
-  BanditPolicyState() = default;
-  ~BanditPolicyState() = default;
-
-  void set_model_map(Map map);
-  void add_model(VersionedModelId id, ModelInfo model);
-  void set_weight_sum(double sum);
-  std::string serialize() const;
-  static BanditPolicyState deserialize(const std::string& bytes);
-  std::string debug_string() const;
-  Map model_map_;
-  double weight_sum_ = 0.0;  // Only for Exp3, Exp4
+  SelectionState() = default;
+  SelectionState(const SelectionState&) = default;
+  SelectionState& operator=(const SelectionState&) = default;
+  SelectionState(SelectionState&&) = default;
+  SelectionState& operator=(SelectionState&&) = default;
+  virtual ~SelectionState() = default;
+  virtual std::string get_debug_string() const = 0;
```

Comment:
It is called from the admin frontend. There's no particular format any policy needs to conform to. Whatever the implementer thinks would be useful information to expose for debugging/runtime inspection.